_docs/modules.md

```diff
@@ -0,0 +1,164 @@
+---
+title: Separating your code into Modules
+category: Imports and Modules
+---
+
+Modules in Glu allow you to organize and reuse code across different files and projects. Glu provides flexible import mechanisms that allow you to control how external files and their content are imported. This flexibility is essential for managing dependencies, keeping your code clean, and avoiding name conflicts.
+
+In this article, we'll explore different ways to import modules in Glu and discuss why it's important to use them effectively.
+
+## Basic Module Import
+
+In Glu a module is implicitly created for each file. It will include everything contained within the file, and will be named after it.
+
+The simplest way to import a module is using the `import` statement followed by the module's filename.
+
+For example, let's say you have a file called `file.glu`:
+
+```glu
+func function() {
+    std::print("Hello from file!");
+}
+```
+
+To use this function in another file, simply import the file and call the function with the module's name as a prefix:
+
+```glu
+import file;
+
+func main() {
+    file::function(); // This will print "Hello from file!"
+}
+```
+Here, `file::function()` is used to call `function` from the `file` module. The `::` operator is used to access functions, constants, and other definitions from the imported module.
+
+## Renaming Imports
+
+Sometimes, the module name can be long or ambiguous. To make the code more readable, you can rename the imported module using the `as` keyword:
+
+```glu
+import file as toto;
+
+func main() {
+    toto::function();
+}
+```
+In this example, `file` is renamed to `toto`, so you can access the `function()` using `toto::function()`. This is particularly useful when you have multiple modules with similar names or when you want to shorten the module name for convenience.
+
+## Importing Specific Functions
+
+If you only need a specific function from a module, you can import it directly:
+```glu
+import file::function;
+
+func main() {
+    function();
+}
+```
+By importing only `function()`, you don't need to reference the module name when calling the function. This can make the code cleaner, especially when working with frequently used functions.
+
+## Importing Multiple Items with Renaming
+
+You can also import multiple items from a module and rename them individually. This can be useful when you need to resolve naming conflicts or want to clarify the usage of certain functions:
+
+```glu
+import file::{function, function1 as function2};
```

Comment:
you should define `function1` in the `file.glu` above so it can compile if someone wants to try.
Maybe also use a better name than function1 and function2.